src/twisted/mail/imap4.py

```diff
@@ -209,33 +275,76 @@ def clean(self):
         Clean ranges list, combining adjacent ranges
         """
 
-        self.ranges.sort()
+        ranges = sorted((self._swap(low, self._infinity, ifIs=None),
```

Comment:
This same, fairly complex double-swap expression happens both at the top and the bottom of this function. It should be pulled out to a separate method (although I gather the `sorted` and `list` invocations can stay here.)